app/assets/javascripts/ui_components/modal.coffee

```diff
@@ -0,0 +1,13 @@
+$ ->
+  _.each $('.ui-components-modal'), (modal) ->
+    $modal = $ modal
+    trigger = $($modal.attr('data-trigger'))
+    return if trigger.length == 0
+    $(trigger).click (e) ->
+      url = $modal.attr('data-url')
+      if url && url.length > 0
+        $modal.find('.modal-body').load url, ->
+          $modal.one('shown.bs.modal', (e) -> $modal.trigger('uic:domchange'))
```

Comment:
That's not what I meant. You are already in the callback of `jQuery.load`, at that point the DOM should already be there, right? Just hidden.